sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -59,8 +59,8 @@ object Cast {
     case (StringType, TimestampType) => true
     case (BooleanType, TimestampType) => true
     case (DateType, TimestampType) => true
-    case (_: NumericType, TimestampType) => true
-
+    case (_: NumericType, TimestampType) => if ( SQLConf.get.numericConvertToTimestampEnable ) true
```

Comment:
Just `SQLConf.get.numericConvertToTimestampEnable`?

Reply:
yes,it's a flag, we will fail for casting numeric to timestmap by default, if enable, then we provide two choices for user ,or reject the casting